Basket-Donation-Payments/Basket-Donation-Payments/View Controllers/Login & Signup Flow/UserSignupViewController.m

```diff
@@ -36,79 +37,51 @@ - (void)viewDidLoad {
-            UIAlertController *alert = [Utils createAlertControllerWithTitle:@"One or more text field is empty." andMessage:@"Please fill out all required info." okCompletion:nil cancelCompletion:nil];
-            [self presentViewController:alert animated:YES completion:nil];
-        } else {
-            [self createNewUser];
-            if ([segueIdentifierToPerform isEqualToString:@"nonprofitCreationSegue"]) {
-                [self performSegueWithIdentifier:segueIdentifierToPerform sender:nil];
-                
-            } else if ([segueIdentifierToPerform isEqualToString:@"signUpSuccessSegue"]) {
-                self.user.nonprofit = nil;
-                [self.user signUpInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error) {
-                    if (!succeeded) {
-                        UIAlertController *alert = [Utils createAlertControllerWithTitle:@"Could not create user." andMessage:error.localizedDescription okCompletion:nil cancelCompletion:nil];
-                        [self presentViewController:alert animated:YES completion:nil];
-                    } else {
-                        [self performSegueWithIdentifier:segueIdentifierToPerform sender:nil];
-                        
-                    }
-                }];
-            }
-            [self createUserToSave];
-            
-            if ([segueIdentifierToPerform isEqualToString:@"nonprofitCreationSegue"]) {
-                //TODO: verify everything looks good before moving on
-                [self performSegueWithIdentifier:segueIdentifierToPerform sender:nil];
-                
-                
-            } else if ([segueIdentifierToPerform isEqualToString:@"signUpSuccessSegue"]) {
-                self.user.nonprofit = nil;
-                [self.user signUpInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error) {
-                    if (!succeeded) {
-                        UIAlertController *alert = [Utils createAlertControllerWithTitle:@"Could not create user." andMessage:error.localizedDescription okCompletion:nil cancelCompletion:nil];
-                        [self presentViewController:alert animated:YES completion:nil];
-                    } else {
-                        [self performSegueWithIdentifier:segueIdentifierToPerform sender:nil];
-                        
-                    }
-                }];
-            }
+
+        [self createUserToSave];
+        if ([segueIdentifierToPerform isEqualToString:@"nonprofitCreationSegue"]) {
+            [self performSegueWithIdentifier:segueIdentifierToPerform sender:nil];
+
+        } else if ([segueIdentifierToPerform isEqualToString:@"signUpSuccessSegue"]) {
+            self.user.nonprofit = nil;
+            NSString *fullName = [NSString stringWithFormat:@"%@ %@", self.user.firstName, self.user.lastName];
```

Comment:
This should be moved to a method in the User class so it can be reused (in the profile view IIRC)